README.md

```diff
@@ -142,16 +144,91 @@ In order to create **Named credentials**:
 
 ### Specifying credentials in the Apex code
 
-Storing credentials in the Apex code is not recommended. If possible, use **Named Credentials**.
+Setting credentials in the code is always an option, and in fact, it's the only option if you're authenticating with an API key (in the case of the Visual Recognition service) or with IAM.
+
+You can always set these values directly in the constructor or with a method call after instantiating your service.
+
+_Note: You must set the service endpoint manually when setting your credentials this way. Otherwise, the SDK will default to searching for `Named Credentials` that you won't have set up._
 
-However, if you choose to, here's an example of setting the credentials in your code:
+#### Username and Password
 
 ```java
+// in the constructor
+IBMDiscoveryV1 discovery = new IBMDiscoveryV1('2017-11-07', 'USERNAME', 'PASSWORD');
+discovery.setEndPoint('URL');
+```
+
+```java
+// after instantiation
 IBMDiscoveryV1 discovery = new IBMDiscoveryV1('2017-11-07');
 discovery.setEndPoint('URL');
 discovery.setUsernameAndPassword('USERNAME', 'PASSWORD');
 ```
 
+#### API Key
+
+```java
+// in the constructor
+IBMVisualRecognitionV3 visualRecognition = new IBMVisualRecognitionV3('2016-05-20', 'API_KEY');
+visualRecognition.setEndPoint('URL');
+```
+
+```java
+// after instantiation
+IBMVisualRecognitionV3 visualRecognition = new IBMVisualRecognitionV3('2016-05-20');
+visualRecognition.setEndPoint('URL');
+visualRecognition.setApiKey('API_KEY');
+```
+
+#### Using IAM
+
+When authenticating with IAM, you have the option of passing in:
+  - the IAM API key and, optionally, the IAM service URL
+  - an IAM access token
+
+**Be aware that passing in an access token means that you're assuming responsibility for maintaining that token's lifecycle.** If you instead pass in an IAM API key, the SDK will manage it for you.
+
+```java
+// in the constructor, letting the SDK manage the IAM token
+IBMWatsonIAMOptions options = new IBMWatsonIAMOptions.Builder()
+  .apiKey('IAM_API_KEY')
+  .url('IAM_URL') // optional - the default value is https://iam.ng.bluemix.net/identity/token
+  .build();
+IBMDiscoveryV1 service = new IBMDiscoveryV1('2017-11-07', options);
+service.setEndPoint('SERVICE_URL');
+```
+
+```java
+// after instantiation, letting the SDK manage the IAM token
+IBMDiscoveryV1 service = new IBMDiscoveryV1('2017-11-07', options);
```

Comment:
unless this code is intended to run after the snippet prior to this, not sure what `options` would refer to in this context. I am guessing maybe it was intended to be:

```java
IBMDiscoveryV1 service = new IBMDiscoveryV1('2017-11-07');
```